tests/py/test_history.py

```diff
@@ -66,16 +73,13 @@ def test_iter_payday_events(self):
         Payday().start()  # to demonstrate that we ignore any open payday?
 
         # Make all events in the same year.
-        delta = '%s days' % (364 - (now - datetime(now.year, 1, 1)).days)
-        self.db.run("""
-            UPDATE paydays
-                SET ts_start = ts_start + interval %(delta)s
-                  , ts_end = ts_end + interval %(delta)s;
-            UPDATE payments
-                SET timestamp = "timestamp" + interval %(delta)s;
-            UPDATE transfers
-                SET timestamp = "timestamp" + interval %(delta)s;
-        """, dict(delta=delta))
+        if now.month < 2:
+
+            # Above, we created four paydays in sequential weeks, with the
+            # latest being today. We don't want them to go back into the
+            # previous year, so let's shift them forward if it's close.
+
+            self.shift_all_paydays_by('3 months')
```

Comment:
Yes, that's the effect. I think for testing purposes it doesn't matter so much whether it's in the future or the past.